asset/fetcher.go

```diff
@@ -65,6 +68,18 @@ func (h *httpFetcher) Fetch(ctx context.Context, url string, headers map[string]
 		h.URLGetter = httpGet
 	}
 
+	limiter, ok := limiterMap[url]
+	if !ok {
+		// 0.03 = once every ~30 seconds per unique url
```

Comment:
Not a bad idea, but I think this would allow for a higher rate than intended if there were many assets being pulled from the same domain. I would expect the domain to rate limit requests for the entire site, not just a single resource.

Perhaps we should have a global rate limit, or perhaps the rate limit should be per domain.

Reply:
Ah, you want to parse each url for the domain rather than differentiate by the unique url. I'll have to think on this overnight.. I'm wondering if there are weird edge cases (for example the different ways you can format an s3 bucket as a url) that we would have to take into account. Its a great suggestion, just need to experiment with it more first!

Reply:
After some Slack discussion, we decided to set a global rate limit rather than per url or per domain. The global rate limit mimics that of the Github API, ~5000 user to server requests per hour.